lib/dictionaryanalyzer.rb

```diff
@@ -0,0 +1,120 @@
+require 'pry'
+require 'yaml'
+
+class DictionaryAnalyzer
+  attr_accessor :return_values
+  def initialize(dictionary_input)
+    @current_dictionary = dictionary_input
+    word_count(dictionary_input)
+    letter_data(dictionary_input)
+    @return_values = []
+    @match_counter = 0
+  end
+
+  def word_count(input)
+    answer = input.length
+    puts "There are #{answer} words in this dictionary.\n"
+  end
+
+  def letter_data(input)
+    ("a".."z").each do |regex|
+      regex_big = regex.upcase
+      letter_count = (input.join("\n").scan(/^#{regex}/).size + input.join("\n").scan(/^#{regex_big}/).size) #regex hates me letter_counter_helper(input, regex) #
+      puts "There are #{letter_count} words under the letter #{regex}"
+    end
+  end
+
+  def user_search_choices
+      #pick user choice
+    user_choice = "nothing"
+    until user_choice.downcase == "q"
+      puts "You may now search this file's dictionary! If you want to search for an exact match, enter '1', for a partial match, enter '2', '3' if it's the beginning of the word and '4' if it's the end of the word you are looking for! Type 'q' to exit any time!"
+      @return_values = []
+      @match_counter = 0
+      user_choice = gets.chomp
+      exit if user_choice.downcase == "q"
+      puts "Great, now what is your search term?"
+      # binding.pry
+      search_term = gets.chomp
+      if user_choice == "1"
+          exact_matches(search_term)
+      elsif user_choice == "2"
+          partial_matches(search_term)
+      elsif user_choice == "3"
+          begins_with(search_term)
+      elsif user_choice == "4"
+          ends_with(search_term)
+      end
+    end
+  end

```

Comment:
`user_search_choices` is 22 lines long. Maybe refactor the user input and the search if/else statement into separate methods.